### DISCUSSION TITLE
Maven artifacts contain Java 11 bytecode, IntellIj confused.

### DISCUSSION THREAD

**Discussion by jeffreycrump-se** [eclipse-ee4j/jersey#5680] [General]
Hi,

I'm trying to implement a Jetty 12/Jersey embeded web app and my IDE is super confused over bytecode versions of `jetty-server `and `jersey-contain-jetty-http`:

In fact, `org.eclipse.jetty:jetty-server:12.0.8 `shows up as compiled with Java 17 but `org.glassfish.jersey.containers:jersey-container-jetty-http:3.1.7` as compiled with Java 11, so my IDE is getting reeeally confused.

My Maven scripts build fine against the same libraries but I suspect I'm not actually getting the Java 17 classes.

In the IDE the problem shows up with code like this: "Cannot resolve method 'setHandler(JettyHttpContainer)'"

```
Server jettyServer; 
...
jettyServer.setHandler(handler);
``` 

The IDE doesn't see the JDK17 bytes so it's not getting the right interface definitions.

Building the containers/jetty-http project locally and installing that in my .m2 repository fixes the issue in the IDE.   Anyone else experience this?  Or am I just losing my mind?

**Comment by jansupol** [comment]
The Jetty modules are multirelease jars. Basically the code throws exception in the case JDK 11 is used, saying "Hey, Jetty requires JDK 17". With JDK 17, the correct classes are loaded by the JDK. The sources jars contain both JDK 11 (fake) classes as well as the correct classes in the  `META-INF/versions` folder.

It may be possible that IDE does not understand multi-release jars

**Reply by jeffreycrump-se** [reply]
I'm pretty sure that it doesn't :-). Was there any thought into producing two (JDK-specific) jars instead?  Would have saved me a days worth of debugging.

 Just wanted to see if anyone else has run into this.

**Reply by jansupol** [reply]
You can vote for a related issue in the IntelliJ: https://youtrack.jetbrains.com/issue/IDEA-285079/Decompiler-Support-multi-release-Jars. There are multiple similar.

**Reply by jeffreycrump-se** [reply]
Thanks!
